### PR TITLE
Feature: setup parser

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

**Review by Hundraw — DISMISSED** [review]

File: src/parser/TerraformListener.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  }
+
+  addComponent() {
+    this.components.push(this.currentComponent);
+    this.currentComponent = null;
+    this.currentBlockType = null;
+  }
+
+  // Enter a parse tree produced by terraformParser#file.
+  enterFile() {
```
Comment (k-y0u) [inline src/parser/TerraformListener.js:30]: why empty ?

File: src/parser/TerraformParser.js [hunk 2]
```diff
@@ -1,8 +1,139 @@
...
+  getParents(components, child) {
+    return components
+      .filter((component) => {
+        if (!component.definition.isContainer) {
+          return false;
+        }
+
+        return child.attributes
+          .some((attribute) => attribute.definition
+              && attribute.definition.type === 'Reference'
```
Comment (k-y0u) [inline src/parser/TerraformParser.js:87]: indent

File: src/parser/TerraformParser.js [hunk 3]
```diff
@@ -1,8 +1,139 @@
...
+              && attribute.definition.containerRef === component.definition.type
+              && attribute.value === component.id);
+      });
+  }
+
+  /**
+   * Indicate if a component is can have parent.
```
Comment (k-y0u) [inline src/parser/TerraformParser.js:94]: `is can` -> `can`

File: tests/unit/parser/TerraformParser.spec.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+              definition: new ComponentLinkDefinition({
+                attributeRef: 'toChild',
+                sourceRef: 'parent',
+                targetRef: 'child',
+                type: 'Default',
+              }),
+            })]);
+        });
+
+        it('test multiple default link', () => {
```
Comment (k-y0u) [inline tests/unit/parser/TerraformParser.spec.js:119]: multiple -> linkS

File: tests/unit/parser/TerraformParser.spec.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+              definition: new ComponentLinkDefinition({
+                attributeRef: 'fromChild',
+                sourceRef: 'child',
+                targetRef: 'parent',
+                type: 'Reverse',
+              }),
+            })]);
+        });
+
+        it('test multiple reverse link', () => {
```
Comment (k-y0u) [inline tests/unit/parser/TerraformParser.spec.js:160]: Same here

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/TerraformListener.js (hunk 1 above)
Comment (Zorin95670) [reply src/parser/TerraformListener.js:30]: It's generated method from grammar.

File: src/parser/TerraformParser.js (hunk 2 above)
Comment (Zorin95670) [reply src/parser/TerraformParser.js:87]: done

File: src/parser/TerraformParser.js (hunk 3 above)
Comment (Zorin95670) [reply src/parser/TerraformParser.js:94]: done

File: tests/unit/parser/TerraformParser.spec.js (hunk 4 above)
Comment (Zorin95670) [reply tests/unit/parser/TerraformParser.spec.js:119]: done

File: tests/unit/parser/TerraformParser.spec.js (hunk 5 above)
Comment (Zorin95670) [reply tests/unit/parser/TerraformParser.spec.js:160]: done

**Review by Hundraw — DISMISSED** [review]

**Review by k-y0u — DISMISSED** [review]

File: tests/unit/parser/TerraformParser.spec.js [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+
+      describe('Test parse: links', () => {
+        const definitions = new TerraformMetadata({
+          metadata: {
+            aws: JSON.parse(fs.readFileSync('tests/resources/tf/link.json', 'utf8')),
+          },
+        }).getDefinitions();
+        const parser = new TerraformParser(definitions);
+
+        it('test single default link', () => {
```
Comment (k-y0u) [inline tests/unit/parser/TerraformParser.spec.js:251]: From here, the `it()` (all of them) should contain sentence which explains the verification (like above).
For instance:
`it("should parse a default link")`

**Review by k-y0u — APPROVED** [review]

**Pull request merged**